### ISSUE TITLE
bug: progress link in user details API

### ISSUE THREAD

**Issue by shadinaif** [nelc/futurex-openedx-extensions#90]
Progress link in user details API should be set as `null` when the caller (the dashboard user) does not have the necessary role like `staff` or `instructor`

**Comment by OmarIthawi** [comment]
@shadinaif this is conflicting with the https://github.com/nelc/futurex-openedx-extensions/issues/101, please chose either to fix.

**Comment by shadinaif** [comment]
#101 is implemented. We still need this for other roles

**Comment by OmarIthawi** [comment]
Yes, the main annoyance is now fixed. This is now a minor bug if at all.
